Lib/tkinter/__init__.py

```diff
@@ -1385,14 +1385,20 @@ def bind(self, sequence=None, func=None, add=None):
         return self._bind(('bind', self._w), sequence, func, add)
 
     def unbind(self, sequence, funcid=None):
-        """Unbind for this widget the event SEQUENCE. if
-        FUNCID is given, delete the command also."""
-        bound = ''
-        if funcid:
-            self.deletecommand(funcid)
+        """Unbind for this widget the event SEQUENCE.
+        
+        If FUNCID is given, only unbind the function identified with FUNCID.
+        and also delete that command.
+        """
+        if funcid is None:
+            self.tk.call('bind', self._w, sequence, None)
```

Comment:
Hey @terryjreedy `None` as third argument to `bind` does not unbind nothing! Take a look at the doc [https://www.tcl.tk/man/tcl8.6/TkCmd/bind.htm](https://www.tcl.tk/man/tcl8.6/TkCmd/bind.htm). I strongly believe the correct form was the one I wrote earlier. 

```
    def unbind(self, sequence, funcid=None):
        """Unbind for this widget the event SEQUENCE.

        If FUNCID is given, only unbind the function identified with FUNCID.
        and also delete that command.
        """
        funcs = self.tk.call('bind', self._w, sequence, None)
        self.tk.call('bind', self._w, sequence, '')
        if funcid:
            self.deletecommand(funcid)
            keep = '\n'.join(f for f in funcs.split('\n') if not f.startswith(f'if {{"[{funcid}'))
            self.tk.call('bind', self._w, sequence, keep)
```

Reply:
"If script is an empty string then the current binding for sequence is destroyed, leaving sequence unbound.", which is the intent of the first call.  An empty string is returned ("In all of the cases where a script argument is provided, bind returns an empty string.")

"If sequence is specified without a script, then the script currently bound to sequence is returned, or an empty string is returned if there is no binding for sequence. "  This is the intent of the second call.  The explicit None is not needed since it is the default.

The code for _bind treats '' and None very differently.

Reply:
@terryjreedy I don't have a lot of spare time these days. I saw you've now correctly removed the `None`  with the empty `str`, as the documentation was pointing, and I'm happy with it. Eventually now this almost 2yo child bug will be fixed in the coming releases ;-)

How can we request the backport of this PR to 3.7 and 3.8?

Reply:
With labels.  I consider this a bugfix, but as Serhiy is the main tkinter maintainer, will let him decide on backporting.

Reply:
Ok! Thank for your answer and for your time :)